lessons/modelevaluation.ipynb

```diff
@@ -0,0 +1,786 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Evaluating Models\n",
+    "\n",
+    "Goals:\n",
+    "\n",
+    "* Be able to design and carry out tests of model adequacy (goodness of fit), and comparisons between models\n",
+    "\n",
+    "* Understand and be prepared to use the Bayesian Evidence"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Further reading\n",
+    "\n",
+    "* Mackay, Chapters 3 and 28\n",
+    "\n",
+    "* Gelman et al (BDA3), Chapter 6"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Preamble\n",
+    "\n",
+    "You can't do inference without making assumptions.\n",
+    "\n",
+    "$\\longrightarrow$ We must _test_ the hypotheses defined by our models."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Three related but distinct questions come under the heading of **model evaluation**.\n",
+    "1. Does a model describe (fit) the data well?\n",
+    "2. Does a model make accurate predictions about new data?\n",
+    "3. How probable are our models in light of the data?\n",
+    "\n",
+    "Often (2) and (3) are directly related to **model comparison** or **selection**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "**A familiar example:** imagine we have a data set like this\n",
+    "<img src=\"graphics/modelcheck-data.png\" width=60%>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Specifically,\n",
+    "* we have precisely known $x$ values\n",
+    "* we have precisely known, Gaussian errors on $y$\n",
+    "* we're fitting a linear model, $\\bar{y}(x)=b+ax$"
```

Comment:
It's a little bit late for this, but the week 4 problems set the linear model at `a+bx`. Because `b+ax` is completely ridiculous. At some point we should make these consistent, provided that we consistently make them `a+bx`.

Reply:
Actually, in the model evaluation notebook we need $y = m x + b$ for total internal consistency (including plots). I believe $m$ for gradient and $b$ for intercept is fairly common... I agree that we should be consistent within an analysis, but on the other hand, it's no bad thing to switch notation between analyses to keep everyone straight with their assumptions... That is what I am telling myself so that I don't have to remake any figures, of course.